### PR TITLE
fix: select on context for bootstrap status channel write

### LINKED ISSUE #56: Always select on a context when writing to a channel that could block
If you write to a channel that and you think the receiver might not read from it (e.g., the receiver could cancel the request), make sure to select on a channel. Grep the codebase for `<-` and audit all the channel writes that don't select on contexts.

### PR DESCRIPTION
**Pull request by alanshaw**
resolves #56

### CODE REVIEW

File: head/head.go [hunk 1]
```diff
@@ -132,23 +132,40 @@ func NewHead(ctx context.Context, options ...opts.Option) (*Head, chan Bootstrap
 	go func() {
 		// ❓ what is this limiter for?
 		if cfg.Limiter != nil {
-			cfg.Limiter <- struct{}{}
+			select {
+			case cfg.Limiter <- struct{}{}:
```
Comment (daviddias) [inline head/head.go:136]: Do we know what the limiter is for now?

Comment (alanshaw) [reply head/head.go:136]: As far as I can tell, it's a mechanism to stagger bootstrap requests so it doesn't appear like we have 200+ IPFS nodes attacking the bootstrappers.

Comment (alanshaw) [reply head/head.go:136]: https://github.com/ipfs/dht-node/commit/fee98576a6662041003ff64d6196a1c52ced1315 this is all I got, is there anything else to know @Kubuxu?

Comment (Kubuxu) [reply head/head.go:136]: Yup, it is just to limit how many bootstraps happen concurrently. Probably to not overload network stack, I can't recall.

**Pull request merged**